enhancements/kubelet/user-namespaces-support.md

```diff
@@ -0,0 +1,477 @@
+| SeccompProfiles                 | seccompProfile              | No impact if a pod is in a user namespace on this field, but if this field is set to `*` or "empty" then it may be tempting to allow all capabilities, but the privileged escalation piece is another aspect to consider. |
+| AllowedUnsafeSysctls            | sysctls                     | No impact--the handling of sysctls should not change if a pod is in a user namespace.                                                                                                                                     |
+| ForbiddenSysctls                | sysctls                     | No impact--the handling of sysctls should not change if a pod is in a user namespace.                                                                                                                                     |
+
+##### `UserNamespaceLevel` field
+
+Add the following to the [Openshift API](https://github.com/openshift/api/blob/76a71dac36a08eab1b240c6c8d4e39c813b1b12b/security/v1/types.go):
+
+```
+       // userNamespaceLevel determines if the policy allows host users in containers.
+       // Valid values are "AllowHostLevel", "RequirePodLevel", and omitted.
+       // When "AllowHostLevel" is set, a pod author may set `hostUsers` to either `true` or `false`.
+       // When "RequirePodLevel" is set, a pod author must `hostUsers` to `false`.
+       // When omitted, this means no opinion and the platform is left to choose a reasonable default,
+       // which is subject to change over time.
+       // The current default is "AllowHostLevel".
+       // +openshift:enable:FeatureGate=UserNamespacesPodSecurityStandards
+       // +kubebuilder:validation:Enum="AllowHostLevel";"RequirePodLevel";""
+       // +optional
+       UserNamespaceLevel NamespaceLevelType `json:"userNamespaceLevel" protobuf:"bytes,26,opt,name=userNamespaceLevel"`
+...
+// NamespaceLevelType shows the allowable values for the UserNamespaceLevel field.
+type NamespaceLevelType string
+...
+const (
+       // NamespaceLevelAllowHost allows a pod to set `hostUsers` field to either `true` or `false`
+       NamespaceLevelAllowHost NamespaceLevelType = "AllowHostLevel"
+       // NamespaceLevelRequirePod requires the `hostUsers` field be `false` in a pod.
+       NamespaceLevelRequirePod NamespaceLevelType = "RequirePodLevel"
+       // NamespaceLevelEmpty will set to the default, which is currently "AllowHostLevel".
+       NamespaceLevelEmpty NamespaceLevelType = ""
+
+...
+)
+
+```
+
+This field branches from its fellow `AllowHost*` namespace fields to follow current API conventions, which maintain boolean fields are not extendable enough.
+
+##### restricted-v3
```

Comment:
One side-note: we have been going through the exercise for pinning SCCs (i.e. setting the `openshift.io/required-scc` annotation) for all workloads in `openshift-*` namespaces. Many of these instances pin the `restricted-v2` SCC; once this goes in GA, we should consider upgrading these to `restricted-v3`, where applicable.

Is this worth mentioning in this enhancement, @haircommander @deads2k?

Reply:
added that as GA requirement and added a blurb to this section